pkg/services/forwarder/listener_nonwin.go

```diff
@@ -0,0 +1,18 @@
+//go:build !windows
+// +build !windows
+
+package forwarder
+
+import (
+	"net"
+	"os"
+)
+
+func listener(socketPath string) (net.Listener, error) {
+	// remove existing socket file
+	if err := os.Remove(socketPath); err != nil && !os.IsNotExist(err) {
+		return nil, err
+	}
+
+	return net.Listen("unix", socketPath) // override tcp to use unix socket
```

Comment:
I think unix sockets can be used on windows as well, after this change, unix sockets are restricted to non-Windows OSes while current `main` branch allowed this on all OSes. And it looks like the PR is not returning errors on unsupported scenarios (trying to use "npipe" on linux, or "unix" on windows). It might even be silently falling back from "npipe" to "unix" or vice-versa depending on the platform?

Probably better to have `listenerUnix` and `listenerNpipe` methods in non-OS specific files. `ListenNpipe` returns an error on platforms where it's not supported, which gives us the missing error reporting mentioned above.

Reply:
A more minor comment, `listener_unix.go` seems to be a more common naming than `listener_nonwin.go`